src/explore-education-statistics-admin/src/pages/release/data/components/ApiDataSetAutoMappedLocationsTable.tsx

```diff
@@ -1,61 +1,176 @@
+            </p>
+          </VisuallyHidden>
+        </div>
+      </div>
+      {filteredLocationChunks.length > 0 ? (
+        <table className={styles.table} data-testid="auto-mapped">
+          <caption className="govuk-visually-hidden">
+            Table showing all auto mapped locations
+          </caption>
+          <thead>
+            <tr>
+              <th className="govuk-!-width-one-third">Current data set</th>
+              <th className="govuk-!-width-one-third">New data set</th>
+              <th>Type</th>
+              <th>Actions</th>
             </tr>
-          );
-        })}
-      </tbody>
-    </table>
+          </thead>
+          <tbody>
+            {filteredLocationChunks[currentPage - 1].map(
+              ({ candidate, mapping }) => {
+                const isPendingUpdate = !!pendingUpdates.find(
+                  update => update.sourceKey === mapping.sourceKey,
+                );
+                return (
+                  <tr key={`mapping-${mapping.source.code}`}>
+                    <td>
+                      {mapping.source.label}
+                      <br />
+                      <span className={styles.code}>{mapping.source.code}</span>
+                    </td>
+                    <td>
+                      {candidate.label}
+                      <br />
+                      <span className={styles.code}>{candidate.code}</span>
+                    </td>
+                    <td>
+                      <Tag colour="grey">Minor</Tag>
+                    </td>
+                    <td>
+                      {isPendingUpdate ? (
+                        <LoadingSpinner
+                          alert
+                          hideText
+                          inline
+                          size="sm"
+                          text="Updating location mapping"
+                        />
+                      ) : (
+                        <ApiDataSetLocationMapModal
+                          candidate={candidate}
+                          level={mapping.level}
+                          mapping={mapping}
+                          newLocations={newLocations[mapping.level]}
+                          onSubmit={onUpdate}
+                        />
+                      )}
+                    </td>
+                  </tr>
+                );
+              },
+            )}
+          </tbody>
+        </table>
+      ) : (
+        'No results found.'
+      )}
+      <Pagination
+        currentPage={currentPage}
+        totalPages={totalPages}
+        renderLink={({
+          'aria-current': ariaCurrent,
+          'aria-label': ariaLabel,
+          'data-testid': testId,
+          children,
+          className,
+          onClick,
+        }) => (
+          <ButtonText
+            ariaCurrent={ariaCurrent}
+            ariaLabel={ariaLabel}
+            className={`govuk-link ${className}`}
```

Comment:
**nitpick** just wondering if this should potentially be using `classnames` for multiple CSS classes?

Reply:
It could, but it doesn't really need to here as the className returned from the pagination is always present.